### DISCUSSION TITLE
Welcome to zenfs Discussions!

### DISCUSSION THREAD

**Discussion by MatiasBjorling** [westerndigitalcorporation/zenfs#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by shampoo365** [comment]
Hello Matias,

When I read the code of ZenFS, I found there is a ZoneExtend class, what is that used for?
Thanks,

Bo Jiang

**Comment by Edward2k** [comment]
Hello Matias,

I am trying to reproduce the artifacts for your paper "_ZNS: Avoiding the Block Interface Tax for Flash-based SSDs_", but am running into quite some issues with the exact versions of tools. 

You mention that Section 5 that you use RocksDB 6.12, but the [first (public) release of this repo](https://github.com/westerndigitalcorporation/zenfs/releases/tag/v1.0.0) mentions _at least_ RocksDB 6.25.3 is required. If possible, could you link the correct commits of `RocksDB`, `ZenFS` and `LibZBD` so I can reproduce the experimentation as close as possible? If not, a mention of any significant differences (that you know of) between the versions used on the paper vs the official release would be highly appreciated :)

Kind Regards,
   Eduardo

**Reply by yhr** [reply]
Edward, you can use the v1.0.2 branch together with the latest libzbd and version 6.x.x of rocksdb, that should be very close to what we used in the paper experiments. There should be no significant differences.

**Reply by Edward2k** [reply]
Thanks!

**Reply by MatiasBjorling** [reply]
Also - make sure to test with the latest and greatest as well. There's been a lot of improvements to performance since that release.

**Reply by Edward2k** [reply]
I definitely will :D

**Comment by miniedwins** [comment]
Hello Matias,

I got some errors about compiling source code. Please help me.
Thanks,

Edward

### OS INFO ###
Ubuntu : 20.04.1
Kernel : 5.13.0-44-generic

### ERRORS  ###
edwin@ubuntu20:~/Downloads/rocksdb$ sudo DEBUG_LEVEL=0 ROCKSDB_PLUGINS=zenfs make -j48 db_bench install
$DEBUG_LEVEL is 0
$DEBUG_LEVEL is 0
  CC       tools/db_bench.o
  CC       tools/db_bench_tool.o
  CC       tools/simulated_hybrid_file_system.o
  CC       test_util/testutil.o
  CC       cache/cache.o
........
........
........
  CC       db/error_handler.o
  CC       db/event_helpers.o
  CC       db/experimental.o
  CC       db/external_sst_file_ingestion_job.o
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make: *** [Makefile:2426: db/compaction/compaction_picker_universal.o] Error 1
make: *** Waiting for unfinished jobs....
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make: *** [Makefile:2426: db/db_impl/db_impl_experimental.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make: *** [Makefile:2426: db/db_impl/db_impl_write.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.

**Reply by aravind-wdc** [reply]
Hi Edward,
I am guessing here, I could be wrong, but it could be possible that you ran out of memory to run compilation with 48 cores.
Could you run with say "-j4" instead of 48. ?
Also what is the gcc/g++ version you are using(and upgrade to latest, if it is older version) ?

**Reply by miniedwins** [reply]
Hi aravind-wdc
It "-j4" works very well. Thank you for your help. 

Thanks,
Edward

**Comment by miniedwins** [comment]
Hello,
Could I discuss Percona Server for Zenfs? I have a problem with "enable rocksdb engine". 
I follow Percona installation guide to setup for server. When I add the setting "loose-rocksdb-fs-uri=zenfs://dev:nvme0n1" in /etc/mysql/my.cnf and reload mysql service, I can't finish this command "ps-admin --enable-rocksdb -u root -p <password>".

Percona Installation Guide : https://www.percona.com/doc/percona-server/8.0/myrocks/zenfs.html#zenfs

Ubuntu : 20.04.1
Kernel : 5.13.0-30-generic

Thanks,
Edward

**Reply by MaisenbacherD** [reply]
Hi Edward,

What error are you getting when enableing RocksDB?
There is a PR with slight improvements for the Percona Installation Guide: https://github.com/percona/percona-server/pull/4762
In case you want to issue a sysbench workload I can also refere to this documentation commit (PR soon to be created): https://github.com/MaisenbacherD/zonedstorage.io/commit/a7f0b58328205842913d40950f89a9fc73164249

**Reply by miniedwins** [reply]
Hi Dennis,
Pleaase let me to clarify again the error I got. When I execute command "ps-admin --enable-rocksdb -u root -p password", if I add "loose-rocksdb-fs-uri=zenfs://dev:nvme0n1" parameter in "/etc/mysql/my.cnf", the command will be hanged without any response.

If the "loose-rocksdb-fs-uri=zenfs://dev:nvme0n1" is removed from "my.cnf", the command "ps-admin --enable-rocksdb -u root -p <PWD>" will not hang. However,  MySQL server will not apply RockDB engine but turn to default InnoDB engine.

Thanks,
Edward

**Reply by MaisenbacherD** [reply]
How does your `/etc/mysql/my.cnf` look like?

**Reply by miniedwins** [reply]
Hi Dennis,

root@edwin-ubuntu:/home/edwin/Downloads# cat /etc/mysql/my.cnf

[mysqld]
loose-rocksdb-fs-uri=zenfs://dev:nvme0n1

!includedir /etc/mysql/conf.d/
!includedir /etc/mysql/mysql.conf.d/

**Reply by miniedwins** [reply]
Following is my installing setup, please refer to my operation.
Link: https://docs.google.com/document/d/1tTr7qsj9Yx3XdS0jqro5RriVUv1xNQaZGlKRxtgQ02U/edit

Thanks,
Edward

**Comment by Z-Jie2020** [comment]
I have no idea. Can someone help me? 
![image](https://user-images.githubusercontent.com/61263660/195401375-ddce99fe-b785-4d0f-b50b-e271d6221cc6.png)

**Reply by Z-Jie2020** [reply]
I installed, downloaded and compiled it according to the official documentation.

**Reply by Z-Jie2020** [reply]
![image](https://user-images.githubusercontent.com/61263660/195402179-63740b43-8567-49d5-8fa0-ed7111741159.png)

**Reply by MaisenbacherD** [reply]
Hi @Z-Jie2020,
Did you configure the IO scheduler and create a ZenFS filesystem according to the documentation before running `db_bench`?
https://github.com/westerndigitalcorporation/zenfs#configure-the-io-scheduler-for-the-zoned-block-device
https://github.com/westerndigitalcorporation/zenfs#creating-a-zenfs-file-system

**Reply by Z-Jie2020** [reply]
Oh sir thank you. I have solved the problem. I am using zenfs on femu, and I found that every time I restart femu, zenfs needs to be recreated.

**Comment by Z-Jie2020** [comment]
Oh sir thank you. I have solved the problem. I am using zenfs on femu, and I found that every time I restart femu, zenfs needs to be recreated.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Dennis ***@***.***&gt;; 
发送时间: 2022年10月17日(星期一) 下午4:04
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [westerndigitalcorporation/zenfs] Welcome to zenfs Discussions! (Discussion #1)





 
Hi @Z-Jie2020,
 Did you configure the IO scheduler and create a ZenFS filesystem according to the documentation before running db_bench?
 https://github.com/westerndigitalcorporation/zenfs#configure-the-io-scheduler-for-the-zoned-block-device
 https://github.com/westerndigitalcorporation/zenfs#creating-a-zenfs-file-system

**Comment by zhoulening969** [comment]
Hello gentlemen,
I got some errors about compiling source code. If only rocksdb is compiled, there is no problem, but adding the Zenfs plugin will result in an error. Please help me.

Thanks,
Eddy

**OS INFO**
Centos : 7.9.2009(core)
Kernel : 6.3.1-1.el7.elrepo.x86_64

**ERRORS**
[root@localhost rocksdb]# DEBUG_LEVEL=0 ROCKSDB_PLUGINS=zenfs make -j48 db_bench install
$DEBUG_LEVEL is 0
plugin/zenfs/zenfs.mk:37: *** Generating ZenFS version failed。
[root@localhost rocksdb]#

**Reply by cheniujh** [reply]
I've met the same issue, but that was because I didn't pull the source code using the 'git clone' command. Instead, I downloaded the source code in zip format and manually unzipped 'zenfs' to 'rocksdb/plugin'. When I followed the instructions in the README file and used 'git clone' to pull the code, this issue was eliminated.

**Comment by MatiasBjorling** [comment]
Hi Zhou,

Could you please create a new issue with your issue, then we are able to
track it independently of this older issue.

Thank you. Best, Matias

**Reply by zhoulening969** [reply]
Okay, thank you, sir
